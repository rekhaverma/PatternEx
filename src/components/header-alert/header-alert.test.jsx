import React from 'react';
import HeaderAlert from './header-alert.component';

/**
 * @status: Complete
 * @sign-off-by: Alex Andries
 */
describe('<HeaderAlert />', () => {
  let wrapper;
  let props;

  beforeEach(() => {
    props = {
      active: true,
      noInactive: true,
      onClick: jest.fn(),
    };
    wrapper = shallow(<HeaderAlert {...props} />);
  });

  it('should match with snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call onClick when element is clicked', () => {
    wrapper.find('.header1__alert').simulate('click');
    expect(props.onClick.mock.calls.length).toBe(1);
  });
});
