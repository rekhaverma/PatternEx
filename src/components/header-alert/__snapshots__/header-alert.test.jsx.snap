// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderAlert /> should match with snapshot 1`] = `
<div
  className="header1__alert"
  onClick={[Function]}
>
  <span
    className="icon-malicious"
  />
  <p />
</div>
`;
